src/devices/cpu/tlcs900/tmp94c241.cpp

```diff
@@ -0,0 +1,1935 @@
+		case 0x7: if ((m_port_function[PORT_C] & 0x03) == 0x03) port_w<PORT_C>(m_port_latch[PORT_C] ^ 0x01); break;
+		case 0x8: if ((m_port_function[PORT_E] & 0x01) == 0x01) port_w<PORT_E>(m_port_latch[PORT_E] ^ 0x01); break;
+		case 0xa: if ((m_port_function[PORT_E] & 0x10) == 0x10) port_w<PORT_E>(m_port_latch[PORT_E] ^ 0x10); break;
+		case 0xb: if ((m_port_function[PORT_C] & 0x03) == 0x03) port_w<PORT_C>(m_port_latch[PORT_C] ^ 0x02); break;
+		}
+		break;
+	case FF_SET:
+		switch( flipflop )
+		{
+		case 0x1: if ((m_port_function[PORT_C] & 0x03) == 0x02) port_w<PORT_C>(m_port_latch[PORT_C] & 0x01); break;
+		case 0x3: if ((m_port_function[PORT_C] & 0x03) == 0x02) port_w<PORT_C>(m_port_latch[PORT_C] & 0x02); break;
+		case 0x4: if ((m_port_function[PORT_D] & 0x01) == 0x01) port_w<PORT_D>(m_port_latch[PORT_D] & 0x01); break;
+		case 0x6: if ((m_port_function[PORT_D] & 0x10) == 0x10) port_w<PORT_D>(m_port_latch[PORT_D] & 0x10); break;
+		case 0x7: if ((m_port_function[PORT_C] & 0x03) == 0x03) port_w<PORT_C>(m_port_latch[PORT_C] & 0x01); break;
+		case 0x8: if ((m_port_function[PORT_E] & 0x01) == 0x01) port_w<PORT_E>(m_port_latch[PORT_E] & 0x01); break;
+		case 0xa: if ((m_port_function[PORT_E] & 0x10) == 0x10) port_w<PORT_E>(m_port_latch[PORT_E] & 0x10); break;
+		case 0xb: if ((m_port_function[PORT_C] & 0x03) == 0x03) port_w<PORT_C>(m_port_latch[PORT_C] & 0x02); break;
+		}
+		break;
+	case FF_CLEAR:
+		switch( flipflop )
+		{
+		case 0x1: if ((m_port_function[PORT_C] & 0x03) == 0x02) port_w<PORT_C>(m_port_latch[PORT_C] | ~0x01); break;
+		case 0x3: if ((m_port_function[PORT_C] & 0x03) == 0x02) port_w<PORT_C>(m_port_latch[PORT_C] | ~0x02); break;
+		case 0x4: if ((m_port_function[PORT_D] & 0x01) == 0x01) port_w<PORT_D>(m_port_latch[PORT_D] | ~0x01); break;
+		case 0x6: if ((m_port_function[PORT_D] & 0x10) == 0x10) port_w<PORT_D>(m_port_latch[PORT_D] | ~0x10); break;
+		case 0x7: if ((m_port_function[PORT_C] & 0x03) == 0x03) port_w<PORT_C>(m_port_latch[PORT_C] | ~0x01); break;
+		case 0x8: if ((m_port_function[PORT_E] & 0x01) == 0x01) port_w<PORT_E>(m_port_latch[PORT_E] | ~0x01); break;
+		case 0xa: if ((m_port_function[PORT_E] & 0x10) == 0x10) port_w<PORT_E>(m_port_latch[PORT_E] | ~0x10); break;
+		case 0xb: if ((m_port_function[PORT_C] & 0x03) == 0x03) port_w<PORT_C>(m_port_latch[PORT_C] | ~0x02); break;
+		}
+		break;
+	}
+}
+
+void tmp94c241_device::t02ffcr_w(uint8_t data)
+{
+	switch( data & 0x0c )
+	{
+	case 0x00:
+		change_timer_flipflop( 1, FF_INVERT );
+		break;
+	case 0x04:
+		change_timer_flipflop( 1, FF_SET );
+		break;
+	case 0x08:
+		change_timer_flipflop( 1, FF_CLEAR );
+		break;
+	}
+	switch( data & 0xc0 )
+	{
+	case 0x00:
+		change_timer_flipflop( 3, FF_INVERT );
+		break;
+	case 0x40:
+		change_timer_flipflop( 3, FF_SET );
+		break;
+	case 0x80:
+		change_timer_flipflop( 3, FF_CLEAR );
+		break;
+	}
+
+	SFR_T02FFCR = data | 0xcc;
+}
+
+void tmp94c241_device::treg23_w(offs_t offset, uint8_t data)
+{
+
+    if (BIT(0, offset))
```

Comment:
This should be `BIT(offset, 0)` ?

Reply:
Yes, indeed. I've just pushed a commit fixing it in both places.